st2-pastebin/sensors/pastebin_poller.py

```diff
@@ -0,0 +1,135 @@
+    def poll(self):
+        """ does the polling action
+            if it finds a "new" paste, emits a payload:
+                {
+                    'date' : int,
+                    'key' : str,
+                    'title' : str,
+                    'user' : str,
+                    'size' : int,
+                    'syntax' : str,
+                }
+        """
+        self.set_poll_interval(self._config['poll_interval'])
+        self._logger.debug('set poll interval to {}'.format(self._config['poll_interval']))
+        
+        
+        try:
+            # do the HTTP request
+            self._logger.debug("Doing the request to {}".format(self._url))
+            req = request_get_versioned(self._url, self._config['ipversion'])
+
+            if req and req.status_code == 200:
+                self._logger.debug("Got a response from the API")
+                try:
+                    jsondata = req.json()
+                except TypeError:
+                    self._logger.debug("JSON Decode failed, stopping. Probably not running from a whitelisted IP")
+                    return
+                if "VISIT: https://pastebin.com/doc_scraping_api TO GET ACCESS!" in req.text:
+                    self._logger.debug("Our source IP is not whitelisted, stoppping.")
+                    return
+                # sort by timestamp, it comes in most-recent-first
+                data = sorted(jsondata, key=lambda k: k['date'])
+                for paste in data:
+                    self._logger.debug("time:{} key:{}".format(paste['date'], paste['key']))
+                    if paste['date'] > self._get_last_time():
+                        # this is the timestamp of the last processed paste
+                        self._set_last_time(last_time=paste['date'])
+                        # do the thing
+                        payload = {'date' : int(paste['date']), 
+                                    'key' : paste['key'],
+                                    'size' : int(paste['size']),
+                                    'user' : paste['user'],
+                                    'title' : paste['title'],
+                                    'syntax' : paste['syntax'],
+                                    }
+                        self._sensor_service.dispatch(trigger=self._trigger_ref, payload=payload)
+                    else:
+                        self._logger.debug("Skipping paste {} {}".format(paste['key'], paste['date']))
+            else:
+                self._logger.debug("No response from the API (status_code: {})".format(req.status_code))
+        except Exception as error_message:
+            self._logger.debug("Threw an error: {}".format(error_message))
+            self._logger.debug(traceback.format_exc())
+        return
+
+    def _get_last_time(self):
+        """ returns the last timestamp that was processed by the poller """
+        if not self._last_time and hasattr(self._sensor_service, 'get_value'):
```

Comment:
these functions should always exist on the sensor_service, any reason you're checking for them?